tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.server.ft.journal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import alluxio.grpc.table.Layout;
+import alluxio.grpc.table.PartitionInfo;
+import alluxio.grpc.table.Schema;
+import alluxio.grpc.table.Storage;
+import alluxio.master.LocalAlluxioCluster;
+import alluxio.master.table.AlluxioCatalog;
+import alluxio.master.table.Table;
+import alluxio.master.table.TableMaster;
+import alluxio.table.common.layout.HiveLayout;
+import alluxio.table.common.udb.UdbConfiguration;
+import alluxio.table.common.udb.UdbContext;
+import alluxio.table.common.udb.UdbTable;
+import alluxio.table.common.udb.UnderDatabase;
+import alluxio.table.common.udb.UnderDatabaseRegistry;
+import alluxio.table.under.hive.HiveDatabaseFactory;
+import alluxio.table.under.hive.HivePartition;
+import alluxio.testutils.LocalAlluxioClusterResource;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Integration tests for table master functionality.
+ */
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.crypto.*", "javax.security.*", "sun.security.*", "javax.net.ssl.*"})
+@PrepareForTest(AlluxioCatalog.class)
+public class TableMasterJournalIntegrationTest {
+  @Rule
+  public LocalAlluxioClusterResource mClusterResource =
+      new LocalAlluxioClusterResource.Builder().setStartCluster(false)
+          .build();
+
+  private static final String DB_NAME = "testdb";
+  private static final String TABLE_NAME_PREFIX = "testtable";
+
+  private void createMockUdbAndTable(int numOfTables) throws Exception {
```

Comment:
would it be simpler to just modify the udb registry class to allow tests to set entries in the registry? Or, be able to pass in a registry to the master classes?

Reply:
the difficulties with that is all the journal operation happens when TableMaster class is started. That's why I mocked the constructor.  Passing in a registry would mean a significant change to `LocalAlluxioClusterResource` and the startMasters process.. which is not ideal either.